hs_core/hydroshare/hs_bagit.py

```diff
@@ -41,217 +50,104 @@ def delete_files_and_bag(resource):
     b = Bags.objects.create(
         content_object=resource.baseresource,
         timestamp=resource.updated
     )
 
+
     return b
 
 
-def read_bag(bag_path):
-    """
-    :param bag_path:
-    :return:
-    """
+def get_remote_file_manifest(resource):
+    json_data = ''
+    data_list = []
+    for f in ResourceFile.objects.filter(object_id=resource.id):
+        data = {}
+        irods_file_name = resource.short_id + "/data/contents/" + f.file_name
+        irods_dest_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+        irods_server_prefix = settings.IRODS_HOST + ':' + settings.IRODS_PORT
+        irods_file_url = 'irods://' + irods_server_prefix +  irods_dest_prefix + "/" + irods_file_name
+        istorage = resource.get_irods_storage()
+
+        tmpdir = os.path.join(settings.TEMP_FILE_DIR, uuid4().hex)
+        tmpfile = os.path.join(tmpdir, f.file_name)
+
+        os.makedirs(tmpdir)
+
+        srcfile = f.reference_file_path
 
-    tmpdir = None
+        if srcfile is None or len(srcfile) == 0:
+            srcfile = os.path.join(irods_dest_prefix, irods_file_name)
 
-    try:
-        if not os.path.exists(bag_path):
-            raise HsBagitException('Bag does not exist')
-        if os.path.isdir(bag_path):
-            unpacked_bag_path = bag_path
+        try:
+            istorage.getFile(srcfile, tmpfile)
+        except:
+            pass
         else:
-            mtype = mimetypes.guess_type(bag_path)
-            if mtype[0] != 'application/zip':
-                msg = "Expected bag to have MIME type application/zip, " \
-                      "but it has {0} instead.".format(mtype[0])
-                raise HsBagitException(msg)
-            tmpdir = tempfile.mkdtemp()
-            zfile = zipfile.ZipFile(bag_path)
-            zroot = zfile.namelist()[0].split(os.sep)[0]
-            zfile.extractall(tmpdir)
-            unpacked_bag_path = os.path.join(tmpdir, zroot)
-
-        bag = bagit.Bag(unpacked_bag_path)
-        if not bag.is_valid():
-            msg = "Bag is not valid"
-            raise HsBagitException(msg)
-
-    finally:
-        if tmpdir:
-            shutil.rmtree(tmpdir)
+            checksum_md5 = mca.compute_checksum(tmpfile, hashlib.md5())
+            checksum_sha256 = mca.compute_checksum(tmpfile, hashlib.sha256())
+            data['url'] = irods_file_url
+            data['length'] = f.size
+            data['filename'] = f.file_name
+            data['md5'] = checksum_md5
+            data['sha256'] = checksum_sha256
+            data_list.append(data)
+
+    return json.dumps(data_list)
+
+def get_metadata_json(resource):
+    data = {}
+    data['BagIt-Profile-Identifier'] = "https://raw.githubusercontent.com/fair-research/bdbag/master/profiles/bdbag-profile.json"
+    data['External-Description'] = "CommonsShare BDBag for resource " + resource.short_id
+    data['Arbitrary-Metadata-Field'] = "TBD Arbitrary metadata field"
+
+    return json.dumps(data)
```

Comment:
You can just return ```data``` so that the calling routine can call json.dump(returned_data, output_file_name) directly to write dict object directly to json file.